src/main/webapp/query-control.js

```diff
@@ -119,4 +119,48 @@ function replaceValueIfEmpty(dataListId) {
   if (typedSoFar === '') {
     inputlist.value = storedVal;
   }
-}
\ No newline at end of file
+}
+
+// Sort location dropdown alphabetically
+function sortStateDropdownList() {
+  // Adapted from w3schools
+  const list = document.getElementById('location');
+  let switching = true;
+  let i;
+  while (switching) {
+    switching = false;
+    options = list.getElementsByTagName('option');
+    // all states option always first
+    for (i = 1; i < (options.length - 1); i++) {
+      shouldSwitch = false;
+      if (options[i].value.toLowerCase() > options[i + 1].value.toLowerCase()) {
+        shouldSwitch = true;
+        break;
+      }
+    }
+    if (shouldSwitch) {
+      options[i].parentNode.insertBefore(options[i + 1], options[i]);
+      switching = true;
+    }
+  }
+}
+
+function createStateDropdownList() {
+  const datalist = document.getElementById('location');
+  let optionElem = document.createElement('option');
+  optionElem.value = 'each U.S. state';
+  optionElem.setAttribute('data-value', 'state');
+  datalist.appendChild(optionElem);
+
+  for (const state in stateInfo) {
```

Comment:
Would it potentially make more sense to sort the states before you iterate through them, so that they are added to the dropdown in sorted order, rather than adding them unsorted and then sorting the dropdown? If you could get them into an array (like just the name or number), then you could use JS's built-in sort method and wouldn't have to write your own.

Reply:
^^ You would have to iterate through the stateInfo file an extra time but having a sorted list of the numerical values of all the states might be useful for some future functions.